migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * name : deleteDuplicateRecords.js
+ * author : Saish Borkar
+ * created-date : 1 Jul 2024
+ * Description : Delete duplicate records having same userId and solutionId to enforce compound key 
+ *               and data integrity
+ */
+
+const path = require("path");
+let rootPath = path.join(__dirname, "../../");
+require("dotenv").config({ path: rootPath + "/.env" });
+const {validate : uuidValidate,v4 : uuidV4} = require('uuid');
+let _ = require("lodash");
+let mongoUrl = process.env.MONGODB_URL;
+let dbName = mongoUrl.split("/").pop();
+let url = mongoUrl.split(dbName)[0];
+var MongoClient = require("mongodb").MongoClient;
+
+var fs = require("fs");
+
+function generateUUId() {
+    return uuidV4();
+  }
+  
+
+(async () => {
+  let connection = await MongoClient.connect(url, { useNewUrlParser: true });
+  let db = connection.db(dbName);
+
+  try {
+    let pipeline = [
+      {
+        $match: {
+          userId: { $exists: true, $ne: null },
+          solutionId: { $exists: true, $ne: null },
+        },
+      },
+      {
+        $group: {
+          _id: { solutionId: "$solutionId", userId: "$userId" },
+          count: { $sum: 1 },
+          docs: { $push: "$$ROOT" },
+        },
+      },
+      {
+        $match: {
+          count: { $gt: 1 },
+        },
+      },
+      {
+        $project: {
+          _id: 0,
+          duplicateCount: "$count",
+          duplicateArray: "$docs",
+        },
+      },
+    ];
+    let duplicateArray = await db
+      .collection("projects")
+      .aggregate(pipeline)
+      .toArray();
+
+    let successfullyDeletedRecords = [];
+    let failedToDeletedRecords = [];
+    for (let duplicateRecord of duplicateArray) {
```

Comment:
create chunk and process each chunk for reference you can check another migration in same repo @borkarsaish65

Reply:
Done this now